DSharpPlus/Entities/Channel/Message/DiscordMessage.cs

```diff
@@ -531,6 +551,28 @@ public Task ModifyEmbedSuppressionAsync(bool hideEmbeds)
         public Task DeleteAsync(string reason = null)
             => this.Discord.ApiClient.DeleteMessageAsync(this.ChannelId, this.Id, reason);
 
+
+        /// <summary>
+        /// Creates a thread. This thread is always a <see cref="ChannelType.PrivateThread"/>
+        /// </summary>
+        /// <param name="name">The name of the thread.</param>
+        /// <param name="auto_archive_duration"><see cref="ThreadAutoArchiveDuration"/> till it gets archived. Defaults to <see cref="ThreadAutoArchiveDuration.OneHour"/></param>
+        /// <param name="reason">The reason.</param>
+        /// <returns></returns>
+        /// <exception cref="Exceptions.UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.UsePrivateThreads"/> or <see cref="Permissions.SendMessages"/> permission.</exception>
+        /// <exception cref="Exceptions.NotFoundException">Thrown when the channel does not exist.</exception>
+        /// <exception cref="Exceptions.BadRequestException">Thrown when an invalid parameter was provided.</exception>
+        /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the <see cref="ThreadAutoArchiveDuration"/> cannot be modified or <see cref="ChannelType.PrivateThread"/> is not enabled for guild. This happens, if the guild does not have <see cref="PremiumTier.Tier_2"/>.</exception>
```

Comment:
"Thrown when the guild does not have the required boost level for the specified <see cref="ThreadAutoArchiveDuration"/>" is more concise.